org.eclipse.xtext/src/org/eclipse/xtext/naming/QualifiedName.java

```diff
@@ -486,7 +494,7 @@ public boolean startsWithIgnoreCase(QualifiedName prefix) {
 	}
 
 	protected boolean startsWith(QualifiedName prefix, boolean ignoreCase) {
-		Preconditions.checkArgument(prefix != null, "prefix must not be null");
+		Objects.requireNonNull(prefix, "prefix must not be null");
```

Comment:
This now throws a `NullPointerException` instead of an `IllegalArgumentException`. I assume that's the reason for the test-failure.
Is that an acceptable change or should it continue to throw a `IllegalArgumentException` if the argument is null?